### PR TITLE
Add YouTube scraper to Hypatia

### PR DESCRIPTION
**Pull request by oneroyalace**
This PR adds the `YoutubeArchiver` gem to Hypatia, enabling it to scrape YouTube videos.

# Testing
1. `bundle install`
 2. `rails t`

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -74,6 +74,10 @@ gem "tzinfo-data", platforms: [:mingw, :mswin, :x64_mingw, :jruby]
 # The whole point of this app
 gem "zorki", "0.1.0", git: "https://github.com/cguess/zorki"
 gem "forki", "0.1.0", git: "https://github.com/oneroyalace/forki"
+gem "youtubearchiver", path: "~/hub/YoutubeArchiver"
```
Comment (cguess) [inline Gemfile:77]: Point toward the git repo, not local

File: app/blueprints/post_blueprint.rb [hunk 2]
```diff
@@ -10,6 +10,8 @@ class PostBlueprint < Blueprinter::Base
       to_return = FacebookPostBlueprint.render_as_hash(post)
     when "Zorki::Post"
       to_return = InstagramPostBlueprint.render_as_hash(post)
+    when "YoutubeArchiver::Video"
+      to_return = YoutubeVideoBlueprint.render_as_hash(post)
```
Comment (cguess) [inline app/blueprints/post_blueprint.rb:14]: Any chance we can rename this to "YouTube" or was this for a specific reason? (name spacing?

**Review by cguess — CHANGES_REQUESTED** [review]
Please point to the repo instead of the local version so this is testable.

File: app/blueprints/post_blueprint.rb (hunk 2 above)
Comment (oneroyalace) [reply app/blueprints/post_blueprint.rb:14]: Oops. To line up with Zenodotus, this should read `YoutubePostBlueprint`, not `YoutubeVideoBlueprint`. Are you suggesting `YoutubeBlueprint` instead? Or suggesting renaming the entire scraping module `Youtube`?

Comment (cguess) [reply app/blueprints/post_blueprint.rb:14]: YouTubePostBlueprint (capital 't')

**Comment by cguess** [comment]
Getting this error when running a test
```
Error:
ScraperControllerTest#test_scraping_a_facebook_video_with_force_works:
NoMethodError: undefined method `[]' for nil:NilClass

      poster = creation_story_object["creation_story"]["comet_sections"]["actor_photo"]["story"]["actors"][0]
                                    ^^^^^^^^^^^^^^^^^^
    app/media_sources/facebook_media_source.rb:64:in `retrieve_facebook_post'
    app/media_sources/facebook_media_source.rb:30:in `extract'
    app/media_sources/media_source.rb:34:in `scrape!'
    app/media_sources/media_source.rb:15:in `scrape'
    app/controllers/scraper_controller.rb:20:in `scrape'
    test/controllers/scraper_controller_test.rb:130:in `block (2 levels) in <class:ScraperControllerTest>'
    test/controllers/scraper_controller_test.rb:129:in `block in <class:ScraperControllerTest>'
```

**Comment by oneroyalace** [comment]
@cguess That test failures should be fixed by https://github.com/TechAndCheck/hypatia/pull/22

File: app/blueprints/post_blueprint.rb (hunk 2 above)
Comment (oneroyalace) [reply app/blueprints/post_blueprint.rb:14]: Resolved per [slack discussion](https://techandcheck.slack.com/archives/CHQAH5FGA/p1649101741310739)

**Review by cguess — APPROVED** [review]

**Pull request merged**